cli/args/mod.rs

```diff
@@ -1595,10 +1591,13 @@ impl CliOptions {
   }
 
   pub fn use_byonm(&self) -> bool {
-    self.enable_future_features()
-  }
+    if self.enable_future_features()
+      && self.node_modules_dir_enablement().is_none()
+    {
+      return true;
+    }
```

Comment:
I don't think this is correct. This will force BYON even if you don't have `node_modules/` dir. That seems unwanted

Reply:
Oh wait, yeah it's not correct because it's not checking if the package.json exists.

Reply:
Fixed to check if package.json exists.